core/utils/hashset/hashset.c

```diff
@@ -45,52 +45,62 @@ hashset_t hashset_create()
     return set;
 }
 
-size_t hashset_num_items(hashset_t set)
-{
+size_t hashset_num_items(hashset_t set) {
     return set->nitems;
 }
 
-void hashset_destroy(hashset_t set)
-{
+void hashset_destroy(hashset_t set) {
     if (set) {
         free(set->items);
```

Comment:
```suggestion
        if (set->items) {
            free(set->items);
        }
```

Reply:
`hashset_destroy` method is called from 3 places in the code
1. reactor_common.c,  method _lf_free_all_tokens line # 435
2. reactor_common.c, method _lf_free_all_tokens line # 446
3. hashset.c, method hashset_create line # 40

First 2 would be fine with this freeing without NULL check, # 3 would throw exception since set->items is NULL as a result of memory allocation failed for set->items

Reply:
Good point!